src/parser.ts

```diff
@@ -70,17 +74,16 @@ export class RoundParser {
     }
 
     private async parseRound(filename: string): Promise<void> {
-        return new Promise<void>(resolve => {
+        return new Promise<void>((resolve) => {
             const logStream = fs.createReadStream(filename);
             logStream.on('data', chunk => {
                 this.rawLogData += chunk;
             }).on('end', () => {
-                resolve();
                 this.doneReading = true;
                 this.parseData();
-            }).on('error', (error) => {
                 resolve();
-                console.log(error);
+            }).on('error', (error) => {
+                console.error(error);

```

Comment:
Is the 'end' event still fired after an error? If not, should we resolve or reject the promise?

Reply:
This would be a filesystem exception (e.g., can't read file), so I didn't really test it.  I'll add a reject call.